### DISCUSSION TITLE
Inconsistent Data

### DISCUSSION THREAD

**Discussion by Alkibiades410** [Stellarium/stellarium#214] [Q&A]
### Expected Behaviour
Consistent Data

### Actual Behaviour
Inconsistent data, which occurs two different ways. 

1) repeat event on PC several days apart.  Clarification: mark event (e.g. Acronycal Setting of Arcturus on 5 May 455 BCE [-454] from N 37 58 55.54 / E 23 44 36.58 @ 300m [daylight savings disabled]), which provides the usual data, but specifically the RA/dec, HA/Dec, Az/Alt change (off by one second +/- to provide the same data); in this case I ran them 3 days apart 

2) repeat event on two different PC's with one running the 32bit version and one running the 64 bit version.

### Steps to reproduce

For #2 above ...

Set location at N 37 58 55.54 / E 23 44 36.58 @ 300m, disable daylight savings time, show constellation lines (no artwork).  Set time at -454/5/28 @ 04:10:06, search for Arcturus (though this also occurs with other stars ... specifically when re-examining Heliacal / Acronycal Risings and Settings, very rarely with solstices/equinoxes).

32bit version reports HA/dec = 7h45m01.69s / +33 36'41.0 || Az/Alt +311 35'51.4 / +2 52'52.5"
64bit version reports HA/dec = 7h45m02.37s / +33 36'41.2 || Az/Alt +311 35'57.5 / +2 52'46.7

RA/dec J2000 and on date report consistent data

Apparent & Mean sidereal time also differ, e.g. Apparent (32bit) 20h08m31.6 vs (64bit) 20h08m32.3s

### System
* Stellarium version: 18.1 (this also occurred under 17.0 & 18.0)
* Operating system: Windows 7 (32 & 64 bit) SP 1
* Graphics Card: 32bit: Intel HD 8.15.10.2189 / 64 bit: Intel HD 5500 20.19.15.4835

### Logfile 
Will supply if needed.

Many thanks in advance.  You lot do a great job, and I couldn't finish my book without this program.

**Comment by Alkibiades410** [comment]
oh ... forgot ... both versions use Espenak & Meeus (2006) for Delta-T; both PC's use the system time (on a domain using the same time-server).

**Comment by gzotti** [comment]
The sub-second difference is not very dramatic, and it even may be expected, depending on how you operate Stellarium.

Some computations go with a 1-second step, and it is likely that the absolute time values for those are different on the 2 computers (I assume they depend on the sub-second of when you launched the program). Can you set the same time via script: 

http://stellarium.org/doc/0.18/classStelMainScriptAPI.html#a1d3456a9216a7dcf1dfd15ef7114ef4c

Stop the animation time progress, then open script console (F12) and write

core.setDate("-454-05-05T05:45:00", "utc");

Execute with the arrow button.

Now both PCs should give the same numerical data, but I cannot exclude some functions would yield different rounding somewhere, this might need to be addressed. I don't have a 32bit WIndows system to test. (Maybe my RaspberryPi3, though.) 

Also make sure to use the same atmosphere parameters (esp. pressure&temperature), or switch off atmosphere to compare altitudes.

Hint: For historical things, you should not just disable daylight saving time, but also enable Local Mean (or even True) Solar Time, as this was the "natural" time for the longitude until the 1890s or so.

**Comment by Alkibiades410** [comment]
Thank-you for the prompt reply!  I will include a note about "sub-second rounding."  The issue only really surfaces with a small handful of events during a 70 year period I am recreating (462-392 bce), typically with Heliacal Rising and Settings of stars, when calculated (actual) visibility is <30 sec -- though it has become consequential with a small number of Acronycal Risings (i.e., when it becomes visible ridiculously close to the horizon).

Thanks for the tips, I will incorporate them immediately.

**Comment by gzotti** [comment]
In case you are dealing with the Heliacal rising result from the visibility plugin: I am somewhat unsure if it provides accurate results, I have not verified this. This topic is heavily discussed, and you should know the code used to compute the dates. I would like to see several existing models in parallel here: at least Arcus Visionis vs. Schaefer, and I think for AV there are several versions/criteria by a few authors. Please get at least independent estimates from other programs or own calculation. And in any case, give or take a minute or two for any rising or setting times, never believe the seconds, the real atmosphere cannot be predicted that accurately.

**Comment by Alkibiades410** [comment]
Thanks again for the attention and comments.

I'll clarify what I am doing.  I am attempting to recreate the Ancient Athenian Calendar, which means I am going through each event and each day one-by-one.

You can view the review draft here:  

https://www.academia.edu/s/fefbcc1b33/the-athenian-year-primer-attic-time-reckoning-and-julian-dates-draft2

Second, my operating definitions:

Cosmical Rising (Morning):  when a star ascends above horizon at the same time the Sun rises. 
Cosmical Setting (Vespertinal):  when a star descends below the horizon at the time the Sun sets.

Acronychal Rising (Vespertinal):  when a star has remained below the horizon during the day but has now ascended above the horizon at the time the Sun sets, i.e. the first evening a star becomes visible already above the horizon. 

Acronychal Setting (Morning): when a star has remained above the horizon in the early morning and first visibly descends below horizon at the time the Sun rises – i.e., the star had not set before the Sun rose the previous morning.

Heliacal Rising (Morning): a star has remained below the horizon and experiences its first visible (and quite brief) ascent above the horizon before the morning twilight renders it invisible.

Heliacal Setting (Vespertinal):  a star has remained above the horizon and experiences its final visible descent below the horizon in the evening twilight – i.e., the following evening the star will drop below the horizon while obscured by sunlight."

*******************

One important (if not critical) reason I chose Stellarium remains the landscape function, and, while it took some serious work, I opine I have successfully integrated the actual Athenian horizon as seen from Lycabettus Hill, Greece thus giving me, as much as possible, the actual visible horizon.

Depending on the height of the surrounding hills, the dates of the Observability Plug-in have proven three to four days "off."

I have assumed the plug-in a) calculates from ground level (42 m), whereas Lycabettus Hill is offset by 258 m (+300m) and b) Lycabettus is N 37 58 55.54 / E 23 44 36.58; and c) Observability uses the geometric horizon and "guestimates" from there [now, that said, I am probably wrong on all of those observations].

While I use the "observability Plug-in" for guidance (get me in the general time frame), I manually recreate each of the above.

For example, my initial run showed the (apparent) heliacal rising of Arktouros (α Boo) on 18 Sept 463 (-462) actually occurs at Az. (+) 47˚34'01.6" / Alt. (+) 2˚10'10.6".

Now, I am going back through everything with implementing the settings you suggested (esp. true solar time).  My spreadsheets are proving off by about 4 - 6 minutes, which I am correcting.

For example, (I haven't done Heliacal Risings but) the my initial run for the Acronycal Setting of Arktouros (α Boo) occurred on 29 May 462 bce (-461) occurred at Az. (+) 311˚40'05.8" / (+) 2˚52'21.0" at 4:07:18 AM, which I have now recalculated at Az. (+) 311˚38'10.2" / Alt. (+) 2˚54'10.4" at 4:15:16 AM.

This is a VERY long winded way of saying I am using the observability plug-in but then go and actually recreate using the horizon I created with the panoramic of Athens from Lycabettus ... so I guess I am relying more on the equations for calculating general star visibility than anything else.

**Comment by Alkibiades410** [comment]
Oh, I have already anticipated some of the objections to which you allude here:  accuracy of the conditions, atmospheric distortion, blah blah.  I have also already added a disclaimer / clarification on the "rounding errors" you first mentioned.  In addition, I have written the following:

"Through the blessings of the Goddess Tyche, the center of Nephelokokkugia-A rests 37˚59′1.7160″ (37.983810) N and 23˚43′39.1404″ (23.727539) E at an elevation of 236 feet – precisely where Athens, Greece resides.

Through the miracle of scientific illiteracy,  Nephelokokkugia-A possessed optimal weather conditions at all times from 462/1 BCE to 392/1 BCE:  clear skies, zero light pollution, and only the most minimal atmospheric distortion – i.e., constant temperature, no exceptional water vapor, fog, etc.  Thus, subsequent observations possessed the exact same conditions as the original observation.

Through the miracle of historical illiteracy, Nephelokokkugia-A never experienced war, revolution, drought, or pestilence.  Our resident Ancient Astronomers, moreover, observed all necessary events every day without fail or error, and their records survived whole and intact.  

Through the twin miracles of anthropological and archaeological illiteracy, the Ancient Greeks of Nephelokokkugia-A had all the necessary tools and skills to measure everything in the sky accurately."

**Comment by gzotti** [comment]
The obs. plugin does not take any landscape into account and works with the math. horizon or an altitude which you can set in its config panel. Terminology around Heliacal/cosmic/acronychal was from Meeus Morsels, and other authors may have different/reversed names for some events. Currently I expect deviations when compared with the frequently discussed and usually applied methods from either Ptolemy/Schoch (Arcus Visionis) or Schaefer, both still await implementation. 

If you cannot do own measurements or photographs, you can create a fairly nice panorama with Google Earth (add 3D buildings for the fine-mesh aerial terrain) and my http://homepage.univie.ac.at/Georg.Zotti/php/panoCam.php, although I don't know where on this very steep hill I should prefer to stand, on one of the coffee terraces, the rusty circle, or on the church roof? Your coordinates seem slightly off. 

For your screenshots, you may want to disable oculars plugin to avoid showing the top-right buttons.

**Comment by Alkibiades410** [comment]
Yes.  I consulted three sources and found four differing sets of definitions for the six (6) risings and settings.  I have thus "overruled" and rewritten the definitions of risings and settings. 

I posted the new "definitions" under the session for the book's review on Academia.edu; i.e. the text in the book is now junk -- toss it to the wind.

The inference here is that Meton (and ancient Greek Astronomers) used the whole what is now the Mount Lycabettus Hill "Viewing Area," which will include the Church of St. George.  

The panorama I have was created from the latter. 

The important coordinates (for Lykabettos Hill) I used are N 37 58 55.54 / E 23 44 36.58 -- the decimal equivalents are place holders, as I was writing the manuscript and not focusing on the actual data (so I guess I better correct those now).

Okay ... 37.9819101 vs 37.9819444 || 23.7433988 vs. 23.7433333

Specifically, the former coordinates are the church and the latter coordinates should place us just to the rear of the Church (where the panorama was taken).

For the book to go to press (that is to construct the calendars you see on page 124), I need to verify the 1st visible crescents of the Moon (around) the Summer, Winter Solstices / Vernal, Autumnal Equinoxes; then the Acronycal Rising & Setting as well as the Heliacal Rising of Arktouros (α Boo); and the Acronycal Rising (Sargas [Girtab]) = Scorpio.

I am also producing and will publish a supplement, which will re-calculate the data presented as well as expand it to include all "key" astronomical data from 462/1-393/2 (I use approximately 24 star / star group risings and settings).  

In other words, I am producing full calendars for every year covered (similar to what you see p 57ff).

I will definitely use what you recommend on the third run -- since nadir and azimuth will undoubtedly prove for more accurate then what I have done to align the panorama I have.  The gods willing, I will also have time produce the deviations using DE431.

Once again, I deeply appreciate you suggestions and taking the time to help with this.
